src/components/MyListComponent.jsx

```diff
@@ -1,28 +1,38 @@
 import React, { useEffect, useState } from "react";
 import { Card } from "semantic-ui-react";
 import MyRequestCard from "./MyRequestCard";
-import { getMyRequests } from "../modules/getRequests";
+import { getMyRequests, getMyQuests } from "../modules/getRequests";
 
-const MyListComponent = () => {
-  const [myRequests, setMyRequests] = useState([]);
+const MyListComponent = ({page, status }) => {
+  const [myList, setMyList] = useState([]);
 
   const getList = async () => {
-    const requests = await getMyRequests();
-    setMyRequests(requests);
+    if (page === "requests") {
+      const requests = await getMyRequests();
+      setMyList(requests);
+    } else {
+      const quests = await getMyQuests();
+      setMyList(quests)
+    }
   };
 
   useEffect(() => {
     getList();
   }, []);
 
-  const cards = myRequests.map((request) => (
+  const cards = myList.filter(req => req.status === status).map((request) => (
     <MyRequestCard key={request.id} request={request} />
   ));
+    
+  const message = cards.length === 0 && <h4 id="no-requests-message">The are no {status} {page} to show!</h4>
```

Comment:
We should probably add something like `{ page === "request" ? "reQuests" : "Quests" }` here, so that the message displays correctly

Reply:
Although it actually works grammatically as it is actually